### PR TITLE
Release 4.18.0

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Prepare next release

### Motivation
Especially propagate new keys

### Describe how to test/QA your changes
 * [FEATURE] Update recipes to support amazonlinux 2023 installation [#896]
 Test chef installation on AmazonLinux 2, 2022 and 2023

 * [FEATURE] Add agent configuration value to config template [#888]
 Check inventories_configuration_enabled is in configuration file, and can be enabled with chef configuration 

 * [FEATURE] Actually enforce `apt` and `yum` versions in the `metadata.rb` file [#874]
 Validate below logic
```
# Use '< 6.0.0' with Chef < 12.9
if current_chef_version < Gem::Version.new(12.9)
  depends 'apt', '< 6.0'
else
  depends 'apt'
end

# Use '< 5.0' with Chef < 12.14
if current_chef_version < Gem::Version.new(12.14)
  depends 'yum', '>= 3.0', '< 5.0'
else
  depends 'yum', '>= 3.0'
end
```
 * [FEATURE] support weblogic integration [#891]
 Enable weblogic and check it's enabled in configuration

 * [FEATURE] Add new APT and RPM signing keys [#890]
 Check the new key is downloaded during installation process

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Changes
 =======
 
+# 4.18.0 / 2023-07-04
+ * [FEATURE] Update recipes to support amazonlinux 2023 installation [#896]
+ * [FEATURE] Add agent configuration value to config template [#888]
+ * [FEATURE] Actually enforce `apt` and `yum` versions in the `metadata.rb` file [#874]
+ * [FEATURE] support weblogic integration [#891]
+ * [FEATURE] Add new APT and RPM signing keys [#890]
+ * [BUGFIX] Fix dead link in http_check.rb recipe [#887]
```
Comment (amenasria) [inline CHANGELOG.md:10]: Shouldn't we add these lines' author?

Comment (amenasria) [reply CHANGELOG.md:10]: Also, for this, you'll have to add yourself at the end of the file (contributor's link to their profile)

Comment (chouetz) [reply CHANGELOG.md:10]: Let's do it now and we'll decide if we keep this forever

**Pull request merged**